detector/detector.go

```diff
@@ -76,33 +78,38 @@ func IsKeyNotFound(item *ast.ObjectItem, k string) bool {
 
 func (d *Detector) Detect() []*issue.Issue {
 	var issues = []*issue.Issue{}
+	modules := d.EvalConfig.ModuleConfig
 
-	for ruleName, detectorMethod := range detectors {
+	for ruleName, creatorMethod := range detectors {
 		if d.Config.IgnoreRule[ruleName] {
 			d.Logger.Info(fmt.Sprintf("ignore rule `%s`", ruleName))
 			continue
 		}
 		d.Logger.Info(fmt.Sprintf("detect by `%s`", ruleName))
-		method := reflect.ValueOf(d).MethodByName(detectorMethod)
+		creator := reflect.ValueOf(d).MethodByName(creatorMethod)
+		detector := creator.Call([]reflect.Value{})[0]
+		method := detector.MethodByName("Detect")
 		method.Call([]reflect.Value{reflect.ValueOf(&issues)})
 
-		for name, m := range d.EvalConfig.ModuleConfig {
+		for name, m := range modules {
 			if d.Config.IgnoreModule[m.Source] {
 				d.Logger.Info(fmt.Sprintf("ignore module `%s`", name))
 				continue
 			}
 			d.Logger.Info(fmt.Sprintf("detect module `%s`", name))
-			moduleDetector := &Detector{
-				ListMap:   m.ListMap,
-				Config:    d.Config,
-				AwsClient: d.AwsClient,
-				EvalConfig: &evaluator.Evaluator{
-					Config: m.Config,
-				},
-				Logger: d.Logger,
-				Error:  false,
+			moduleDetector, err := NewDetector(m.ListMap, d.Config)
```

Comment:
__[gometalinter]__
* [gotype] cannot use m.ListMap (variable of type map[string]*github.com/hashicorp/hcl/hcl/ast.ObjectList) as map[string]*invalid type value in argument to NewDetector

<!-- Comment by [SideCI](https://sideci.com) -->